tap_recharge/streams.py

```diff
@@ -0,0 +1,452 @@
+    params = {
+        'sort_by': f'{replication_key}-asc',
+        'owner_resource': 'store'
+        }
+    data_key = 'metafields'
+
+
+class MetafieldsCustomer(PageBasedPagingStream):
+    """
+    Retrieves customer metafields from the Recharge API.
+
+    Docs: https://developer.rechargepayments.com/#list-metafields
+    """
+    tap_stream_id = 'metafields_customer'
+    key_properties = ['id']
+    path = 'metafields'
+    replication_key = 'updated_at' # pseudo-incremental; doesn't support `updated_at_min` param
+    valid_replication_keys = ['updated_at']
+    params = {
+        'sort_by': f'{replication_key}-asc',
+        'owner_resource': 'customer'
+        }
+    data_key = 'metafields'
+
+
+class MetafieldsSubscription(PageBasedPagingStream):
+    """
+    Retrieves subscription metafields from the Recharge API.
+
+    Docs: https://developer.rechargepayments.com/#list-metafields
+    """
+    tap_stream_id = 'metafields_subscription'
+    key_properties = ['id']
+    path = 'metafields'
+    replication_key = 'updated_at' # pseudo-incremental; doesn't support `updated_at_min` param
+    valid_replication_keys = ['updated_at']
+    params = {
+        'sort_by': f'{replication_key}-asc',
+        'owner_resource': 'subscription'
+        }
+    data_key = 'metafields'
+
+
+class Onetimes(PageBasedPagingStream):
+    """
+    Retrieves non-recurring line items on queued orders from the Recharge API.
+
+    Docs: https://developer.rechargepayments.com/#list-onetimes
+    """
+    tap_stream_id = 'onetimes'
+    key_properties = ['id']
+    path = 'onetimes'
+    replication_key = 'updated_at'
+    valid_replication_keys = ['updated_at']
+    params = {'sort_by': f'{replication_key}-asc'}
+    data_key = 'onetimes'
+
+
+class Orders(CursorPagingStream):
+    """
+    Retrieves orders from the Recharge API.
+
+    Docs: https://developer.rechargepayments.com/#list-orders
+    """
+    tap_stream_id = 'orders'
+    key_properties = ['id']
+    path = 'orders'
+    replication_key = 'updated_at'
+    valid_replication_keys = ['updated_at']
+    params = {'sort_by': f'{replication_key}-asc'}
+    data_key = 'orders'
+
+
+class Products(CursorPagingStream):
+    """
+    Retrieves products from the Recharge API.
+
+    Docs: https://developer.rechargepayments.com/#list-products
+    """
+    tap_stream_id = 'products'
+    key_properties = ['id']
+    path = 'products'
+    replication_key = 'updated_at' # pseudo-incremental; doesn't support `updated_at_min` param
+    valid_replication_keys = ['updated_at']
+    params = {'sort_by': f'{replication_key}-asc'}
+    data_key = 'products'
+
+
+class Shop(FullTableStream):
+    """
+    Retrieves basic info about your store setup from the Recharge API.
+
+    Docs: https://developer.rechargepayments.com/#shop
+    """
+    tap_stream_id = 'shop'
+    key_properties = ['id']
+    path = 'shop'
+    replication_key = 'updated_at'
+    valid_replication_keys = ['updated_at']
```

Comment:
Is this copy pasta? It seems like the sync code only cares about this field if the stream is incremental

Reply:
Correct, I've removed the extra attributes.

The `shop` stream only replicates a single record, so there wasn't a need for the params either.